### ISSUE TITLE
Migrate some rule packages to dependencies

### ISSUE THREAD

**Issue by kouhin** [airbnb/javascript#855]
Sorry for reopening it. #720

I strongly recommend that `eslint-plugin-jsx-a11y` `eslint-plugin-import` and `eslint-plugin-react` should be migrated to `dependencies` from `peerDependencies`.

In npm@3, peerDependencies are not automatically installed anymore, after `npm install eslint-config-airbnb --save-dev`, we must install `eslint-plugin-jsx-a11y` `eslint-plugin-import` `eslint-plugin-react` manually and fix VERSIONs for these three packages one by one to match `eslint-config-airbnb`. 

What's more, after packages updated, we have to find out new dependent versions in `eslint-config-airbnb` package.json, and fix VERSIONs again. It's so annoying.

`peerDependencies` is designed for plugin, see https://nodejs.org/en/blog/npm/peer-dependencies.
Of course `eslint-config-airbnb` is not a plugin for `eslint-plugin-jsx-a11y` or `eslint-plugin-import` or `eslint-plugin-react`. 

`eslint-config-airbnb` won't work without these three packages, so I think these should be added to dependencies.

**Comment by ljharb** [comment]
Due to the way `eslint` is designed, they simply can not be.

I'm aware it's annoying, and `dependencies` is indeed the proper place for it, but this is simply the cost of eslint not supporting plugin composition.
